pandora-common/src/character/assetPreferences.ts

```diff
@@ -23,8 +23,8 @@ export const AssetPreferenceSchema = z.object({
 export type AssetPreference = z.infer<typeof AssetPreferenceSchema>;
 
 export const AssetPreferencesSchema = z.object({
-	attributes: z.record(z.string(), AttributePreferenceSchema).default({}),
-	assets: z.record(AssetIdSchema, AssetPreferenceSchema).default({}),
+	attributes: z.record(z.string(), AttributePreferenceSchema.optional()).default({}),
```

Comment:
why would we want to allow optional here, `undefined` is removed from JSON anyway, and we certainly do not want to store `undefinied` in database, so I think the validation should not accept if a key exists with `undefined` as a value

Reply:
Because then TS correctly marks value gotten from the record as potentially `| undefined`.
It also allows doing `{ ...oldValue, [id]: undefined }`, which works well, but if passed to Zod (which happens during send too), it screams about it.

Reply:
Hmm~ But I guess it is not that difficult to do it differently...

Reply:
the first is unfortunate, I do not like the things around this JS/TS behavior issue
for the second we could just use `omit({ ...oldValue }, id)`

I guess that change is fine, I just do not like it